crates/project/src/module.rs

```diff
@@ -155,14 +207,29 @@ fn find_cycle(
     false
 }
 
+pub const SPEND: &str = "spend";
+pub const CERT: &str = "cert";
+pub const MINT: &str = "mint";
+pub const WITHDRAW: &str = "withdraw";
+pub const VALIDATOR_NAMES: [&str; 4] = [SPEND, CERT, MINT, WITHDRAW];
+
 #[derive(Debug, Clone)]
 pub struct CheckedModule {
     pub name: String,
     pub code: String,
     pub input_path: PathBuf,
     pub kind: ModuleKind,
     pub ast: TypedModule,
-    // pub extra: ModuleExtra,
+    pub extra: ModuleExtra,
+}
+
+impl CheckedModule {
+    pub fn is_library(&self) -> bool {
+        self.ast.definitions().any(|def| match def {
+            Definition::Fn(func_def) => VALIDATOR_NAMES.contains(&func_def.name.as_str()),
```

Comment:
this check is confusing me a little. Can't we instead just check self.kind?

Reply:
It's actually unused. I figured in the end that we could also just generate docs for validators if they are any  🤷‍♀️

So this could/should be removed